src/System/ErrorLogger.cpp

```diff
@@ -0,0 +1,95 @@
+#include "ErrorLogger.h"
+#include <string.h>
+
+#define FORMATTED_MESSAGE_SIZE (sizeof("assert failed: (  )\n\t<,ln > in ") + 2)
+
+ErrorLogger* ErrorLogger::_instance = nullptr;
+ErrorLogger& ErrorUtil = ErrorLogger::instance();
+
+ErrorLogger& ErrorLogger::instance() {
+    if (!_instance) {
+        _instance = new ErrorLogger();
+    }
+    return *_instance;
+}
+
+void ErrorLogger::init(ErrorHandlingBehaviour behaviour) {
+    if (behaviour == Log || behaviour == LogAndPrint) {
+        _openLogFile();
+    }
+    _behaviour = behaviour;
+}
+
+void ErrorLogger::setBehaviour(ErrorHandlingBehaviour behaviour) {
+    switch (behaviour) {
+        case Log:
+            _openLogFile();
+            break;
+        case Print:
+            _closeLogFile();
+            break;
+        case LogAndPrint:
+            _openLogFile();
+            break;
+        default:
+            _closeLogFile();
+            break;
+    }
+    _behaviour = behaviour;
+}
+
+void ErrorLogger::errorAssert(const char* file, int line, const char* func, const char* expression) {
```

Comment:
I would suggest to consider renaming `file` to `fileName` for clarity and also to match the manner in which the related variables are named in `DataLogger`. Although I see C++ uses the __FILE__ macro for the file name, so I see where it comes from

Reply:
yep, ```__FILE__``` captures the path of the source file from which it is used.

Reply:
Hmm I think a more descriptive name might be ```srcfile```.  I can make that change